x-pack/heartbeat/scenarios/basics_test.go

```diff
@@ -34,20 +42,45 @@ func TestSimpleScenariosBasicFields(t *testing.T) {
 				},
 			}), e.Fields)
 
-			// Ensure that all check groups are equal and don't change
-			cg, err := e.GetValue("monitor.check_group")
+			// Ensure that all check groups are equal and don't except across retries
+			cgIface, err := e.GetValue("monitor.check_group")
 			require.NoError(t, err)
-			cgStr := cg.(string)
-			if i == 0 {
-				lastCg = cgStr
-			} else {
-				require.Equal(t, lastCg, cgStr)
+			cg := cgIface.(string)
+
+			var summary *summarizer.JobSummary
+			summaryIface, err := e.GetValue("summary")
+			if err == nil {
+				summary = summaryIface.(*summarizer.JobSummary)
```

Comment:
See other comment about checking that the type assertion succeeded.

Reply:
See my comment on the previous comment